### ISSUE TITLE
StringDecoding error for an mp3 file with no tags

### ISSUE THREAD

**Issue by AndrewRadev** [polyfloyd/rust-id3#60]
I'm having problems parsing this mp3 file (zipped for github): [broken.zip](https://github.com/polyfloyd/rust-id3/files/6003955/broken.zip)

When running the `id3` command on it (https://squell.github.io/id3/), I get no tags:

``` sh-session
% id3 -v broken.mp3
#file	broken.mp3
#tag	ID3v2 3.0
```

I'm actually not sure how I removed them, but I know I did it for testing purposes for my project [id3-image](https://github.com/AndrewRadev/id3-image). My tests were running fine and updating from 0.5.1 to 0.5.3, I started getting this error in the test suite. It's also failing on the latest published version, 0.6.2. Here's some code I'm testing with

``` rust
fn main() {
    let tag = id3::Tag::read_from_path("./broken.mp3").unwrap();
    println!("{:?}", tag);
}
```

This fails with:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: StringDecoding([]): data is not valid utf-8', src/main.rs:2:56
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It's not a huge problem, because I just used a different file to test with that seems to work. I don't know why this one fails, though. There might be something malformed, but it does seem to parse fine in 0.5.1, and I'm not sure what changed. Note that putting "0.5.1" in the cargo file actually seems to install 0.5.3, so to test, you might need to use a local path and switch git tags.

**Comment by polyfloyd** [comment]
Confirmed, will investigate further

**Comment by polyfloyd** [comment]
Ehm, the frame tag that seems to be erroneous is `[ff, fb, e0, 44]`. The ff fb looks a bit like an MPEG header that somehow got scrambled up in the tag. There is not anything to be fixed.

You can however, get the tag with the frames preceding the invalid section with the `id3::Error::partial_tag` field. I will add the ID3 header from the file you attached to the test suite.

**Comment by AndrewRadev** [comment]
The problem seems to be in Audacity. I've noticed a while back that when exporting an mp3 file, it would nuke its id3 tags somehow. I just tried opening a random song with audacity and exporting it, and it breaks in the described way.

The broken tags *can* be parsed by some music programs, for instance smplayer renders the title correctly, and so does an mpd client I use. So they're not removed, but I guess audacity maybe has some bug that makes them unparseable by some programs? Ironically, I used to use this project (in my id3-image program) to embed an image, which would "fix" the mp3 file, but I guess that won't work now.

I'll look into the `partial_tag` part, it should help me avoid the audacity problem, thanks.
